### ISSUE TITLE
Remove annotation share links in LMS context

### ISSUE THREAD

**Issue by robertknight** [hypothesis/product-backlog#843]
_Extracted from 
https://github.com/hypothesis/product-backlog/issues/789#issuecomment-429840443_:

In the context of an LMS assignment when account auto-provisioning is enabled, annotation cards have share links which currently take the user to the URL of the document the user is annotating. However since being able to see an annotation relies on the user and group context provided by the LMS app, the annotations are not visible when following these share links.

We could fix this in a couple of ways:

1. Make the share links take the user to the top-level frame's URL
2. Remove the share links in this context

I would suggest we start with (2) since it is much easier to implement and sharing annotation links is not a critical feature in the LMS context and we may consider (1) at a later stage.

**Comment by jeremydean** [comment]
I just want to plant a little flag here. Sorry if it belongs elsewhere, but this issue touches on some important pieces of educational workflow with Hypothesis more broadly and within the LMS context. 

It has less to do with "share links" as such than with somehow linking directly to annotations in context either from our actual activity pages (not directly relevant here) or whatever @lyzadanger et al. cook up as the new "annotations view" within the LMS.     

1) @katelyn may be able to say more here but I've been noticing a lot of user feedback around broken connections between the groups home activity page and actual documents. Users expect these links to work, but they don't in certain contexts: like, I think, if the document is behind a login. We at least need better messaging here. I think users are sometimes told they don't have access to those annotations which isn't actually true. At best, though, we'd have some way to store whatever information is needed to get a user past those gates. For example, if I'm logged into my course website in another tab, shouldn't I be able to link from a group activity page to a doc within?

2) In the LMS context, I expect teachers may want to link to annotations in context from whatever scoped out-of-context view we end of offering them. For example, I might be grading Rob's annotations on Doc X just looking at his annotations out of context. But I might want to refer to the doc itself or respond directly to his annotation. My expectation, as above, would be that I could link directly from the out of context view to the doc context.

**Comment by robertknight** [comment]
> Sorry if it belongs elsewhere, but this issue touches on some important pieces of educational workflow with Hypothesis more broadly and within the LMS context.

Thanks for the feedback. Here is as good as anywhere.

**Comment by seanh** [comment]
^ In the short term I think we're gonna remove, not fix, the share links, as there is no way (that I'm aware of) to make these links work in the LMS context. They probably will never work inside an LMS, is my guess.

**Comment by robertknight** [comment]
We discussed some possible implementations [on Slack](https://hypothes-is.slack.com/archives/C1M8NH76X/p1542712192002300). We came up with three broad options:

1. Implicitly hide these links in cases where the client is using a login provided by a parent frame, since direct links are not going to work in this case.
2. Explicitly require third party authorities to disable share links in their config.
3. Teach h not to return bouncer links for annotations in the LMS authority. This would probably mean adding some DB entity to represent an authority in h explicitly and then adding a flag to that entity.

We're currently leaning towards (2) as it is simple to implement, easy to remove later in favor of (3) and avoids adding additional implicit behaviours in "third party mode". The client config API (in this case the LMS app) might look like this:

```js
services: [{
   grantToken: "...",
   enableShareLinks: false, // default `true`.
}]
```
